dali/server/daserver.cpp

```diff
@@ -752,6 +766,7 @@ int main(int argc, const char* argv[])
         if (ok) {
             writeSentinelFile(sentinelFile);
             covenMain();
+            initializeMetrics(serverConfig);
             removeAbortHandler(actionOnAbort);
         }
         stopServer();
```

Comment:
@jakesmith would it be suitable to call stopCollecting() after this call to stopServer()?

Reply:
yes, think it should be.
If it isn't I suspect there could be potential problem (crashes) during closedown whilst the sink threads are still running and objects they are reporting on are destructed.

Reply:
@jakesmith The framework only maintains weak pointers to metric objects, so if Dali is shutting down and destroying any metric objects, the framework will detect that the object has been deleted and will delete it from its internal collection.

Reply:
It can still cause a problem - if the framework has a strong pointer to an object, and that object is an instance of a class defined in a dll which is then unloaded.  Then the thread can crash when it calls a virtual method of the class whose definition is no longer in memory.  We have even seen instances of this in the past...

Reply:
That is a danger when using custom metrics and perhaps we should better document the concern. Or, custom metrics should be a temporary solution with an impetus to migrate to framework defined classes.

Reply:
If you add stopCollecting() call, the threads I assume will be stopped, therefore avoiding the problem being discussed.